tests/unit/src/test_conditionSet.cpp

```diff
@@ -28,6 +31,19 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 namespace { // test namespace
 using Conditions = std::map<std::string, int64_t>;
 // #region mock data
+
+int primarySize(const ConditionsStore &store)
+{
+	int size = 0;
+	auto it = store.PrimariesBegin();
+	while(it != store.PrimariesEnd())
+	{
+		++it;
+		++size;
+	}
+	return size;
```

Comment:
You can use `std::distance` here.

Reply:
I added a TODO item for this. (As you noticed before; the iterator that is needed for this is not yet correctly implemented.)